core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ActiveDirectoryUserStoreManager.java

```diff
@@ -245,8 +245,8 @@ public void doUpdateCredential(String userName, Object newCredential, Object old
                 .getUserStoreProperty(LDAPConstants.USER_NAME_ATTRIBUTE);
         // String searchFilter =
         // realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_SEARCH_FILTER);
-        String searchFilter = "(&" + userListFilter + "(" + userNameAttribute + "=" + userName
-                + "))";
+        String searchFilter = "(&" + userListFilter + "(" + userNameAttribute + "=" +
```

Comment:
We have opened a public jira for this [1]. We will discuss this within IS team and fix the JIRA. Ideally webapps should not be able to set these properties.

[1] https://wso2.org/jira/browse/IDENTITY-3509